src/mono/wasm/wasm.proj

```diff
@@ -89,6 +80,82 @@
     </ManagedToNativeGenerator>
   </Target>
 
+  <UsingTask TaskName="Microsoft.WebAssembly.Build.Tasks.EmitWasmBundleFiles" AssemblyFile="$(WasmAppBuilderTasksAssemblyPath)" />
+  <Target Name="GenerateTimezonesArchive" Returns="@(_WasmArchivedTimezones)">
+    <PropertyGroup>
+      <_WasmTimezonesPath>$([MSBuild]::NormalizePath('$(PkgSystem_Runtime_TimeZoneData)', 'contentFiles', 'any', 'any', 'data'))</_WasmTimezonesPath>
+      <_WasmTimezonesBundleSourceFile>$(WasmObjDir)\wasm-bundled-timezones.c</_WasmTimezonesBundleSourceFile>
+      <_WasmTimezonesBundleObjectFile>$(WasmObjDir)\wasm-bundled-timezones.o</_WasmTimezonesBundleObjectFile>
+      <_WasmTimezonesBundleArchive>$(WasmObjDir)\wasm-bundled-timezones.a</_WasmTimezonesBundleArchive>
+      <_WasmTimezonesSourcesRsp>$(WasmObjDir)\wasm-bundled-timezones-sources.rsp</_WasmTimezonesSourcesRsp>
+      <_WasmTimezonesArchiveRsp>$(WasmObjDir)\wasm-bundled-timezones-archive.rsp</_WasmTimezonesArchiveRsp>
+    </PropertyGroup>
+    <ItemGroup>
+      <_WasmTimezonesInternal Include="$(_WasmTimezonesPath)\**\*.*" WasmRole="Timezone"/>
+    </ItemGroup>
+    <GetFileHash Files="@(_WasmTimezonesInternal)">
+      <Output TaskParameter="Items" ItemName="_WasmBundleTimezonesWithHashes" />
+    </GetFileHash>
+    <ItemGroup>
+      <_WasmBundleTimezonesWithHashes Update="@(_WasmBundleTimezonesWithHashes)">
+        <DestinationFile>$(WasmObjDir)\wasm-bundled-$([System.String]::Copy(%(_WasmBundleTimezonesWithHashes.FileHash))).c</DestinationFile>
+        <ObjectFile>$(WasmObjDir)\wasm-bundled-$([System.String]::Copy(%(_WasmBundleTimezonesWithHashes.FileHash))).o</ObjectFile>
+        <RegisteredName>$([MSBuild]::MakeRelative($(_WasmTimezonesPath), %(_WasmBundleTimezonesWithHashes.Identity)).Replace('\','/'))</RegisteredName>
+      </_WasmBundleTimezonesWithHashes>
+    </ItemGroup>
+    <!-- TODO make this incremental compilation -->
+    <EmitWasmBundleFiles
+      FilesToBundle="@(_WasmBundleTimezonesWithHashes)"
+      BundleName="timezones"
+      BundleFile="$(WasmObjDir)\wasm-bundled-timezones.c"
+      RegistrationCallbackFunctionName="mono_wasm_add_bundled_file"
+      >
+    </EmitWasmBundleFiles>
+    <ItemGroup>
+      <WasmBundleTimezonesSources Include="$([MSBuild]::MakeRelative($(WasmObjDir), %(_WasmBundleTimezonesWithHashes.DestinationFile)).Replace('\','/'))" />
```

Comment:
`WasmBundleTimezonesSources` -> `WasmBundleTimezonesSources` - matching the others